client/src/app/__tests__/AppSpec.js

```diff
@@ -2177,7 +2193,6 @@ describe('<App>', function() {
       } = app.state;
 
       // then
-      expect(updatedTab).to.not.eql(tab);
```

Comment:
Why is this needed? Does this even make sense?

Reply:
I missed the line that made this make sense. Here it is:
 
 ```diff

-  expect(updatedTab).to.not.eql(tab);
+  expect(updatedTab).to.include(newAttrs);

```
This was causing the test to fail. Even with the new function I added, the old tab object is updated with the new properties. I'm not sure how we can prevent this from happening easily.

I didn't consider assertion relevant, because the "navigation history" test passes, and the new tab is updated correctly (as proved by the new line added).

But let me know if you have different insight on this.